app/models/author.rb

```diff
@@ -35,6 +37,20 @@ def leap_year?
     LeapYear.new(created_at).leap_year?
   end
 
+  def zodiac
+    Zodiac.for_date(date_of_birth)
+  end
+
+  def age_at_first_release
+    date_of_first_book = books.map do |book|
+      book.published_at
+    end.min
+
+    diff = ((date_of_first_book - date_of_birth).to_i)/365.25
+
+    "About #{ diff.floor } years and #{ (diff.modulo(1)*365.25).round(0) } days"
```

Comment:
given we use `365.25` twice here, I'd be inclined to pop it in a self-documenting local variable. EG:

```ruby
average_number_of_days_in_a_year = 365.25
diff = (date_of_first_book - date_of_birth).to_i / average_number_of_days_in_a_year
```

I'd also be inclined to extract some of the interpolated values in the string, again for the purpose of self documenting and because I personally find calculations in interpolations difficult to read

```ruby
years = diff.floor
days = (diff.modulo(1) * average_number_of_days_in_a_year).round(0)

"About #{ years } years and #{ days } days"
```